src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function summator(...arr) {
+    if (arr.length === 0) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    sum += arr.reduce((a, b) => a + b, 0);
```

Comment:
The use of 'reduce' here is unnecessary. Since 'summator' should only accept a single argument at a time, you can simplify this by adding the argument directly to the 'sum'.